tests/unit/test_clidriver.py

```diff
@@ -737,6 +739,32 @@ def test_idempotency_token_is_not_required_in_help_text(self):
             self.driver.main(['servicecatalog', 'create-constraint'])
         self.assertNotIn('--idempotency-token', self.stderr.getvalue())
 
+    @mock.patch('awscli.clidriver.distro.id', return_value='amzn')
+    @mock.patch('awscli.clidriver.distro.major_version', side_effect='1')
+    def test_user_agent_for_linux(self, *args):
+        driver = create_clidriver()
+        user_agent_extra_pattern = re.compile(
+            r'^.+/[a-z1-9_]+\.amzn\.1$'
+        )
+        self.assertIsNotNone(user_agent_extra_pattern.match(
+            driver.session.user_agent_extra))
+
+    def test_user_agent(self, *args):
+        driver = create_clidriver()
+        user_agent_extra_pattern = re.compile(
+            r'^.+/[a-z1-9_]+(\.[a-z]+)?(\.[1-9]+)?$'
```

Comment:
Are there any distro id's that use numbers or capital letters? We may want to lax that constraint as well.

Reply:
According to the `distro.id()` docs it always lowercase and the only think I missed is underscore.